tracer/src/Datadog.Tracer.Native/debugger_method_rewriter.cpp

```diff
@@ -745,15 +785,225 @@ HRESULT DebuggerMethodRewriter::ApplyMethodProbe(
 
     EHClause exClause = {};
     exClause.m_Flags = COR_ILEXCEPTION_CLAUSE_NONE;
-    exClause.m_pTryBegin = firstInstruction;
+    exClause.m_pTryBegin = tryInstruction->m_pNext;
+    exClause.m_pTryEnd = startExceptionCatch;
+    exClause.m_pHandlerBegin = startExceptionCatch;
+    exClause.m_pHandlerEnd = rethrowInstr;
+    exClause.m_ClassToken = debuggerTokens->GetExceptionTypeRef();
+
+    EHClause finallyClause = {};
+    finallyClause.m_Flags = COR_ILEXCEPTION_CLAUSE_FINALLY;
+    finallyClause.m_pTryBegin = tryInstruction->m_pNext;
+    finallyClause.m_pTryEnd = rethrowInstr->m_pNext;
+    finallyClause.m_pHandlerBegin = rethrowInstr->m_pNext;
+    finallyClause.m_pHandlerEnd = endFinallyInstr;
+
+    newClauses.push_back(beginMethodExClause);
+    newClauses.push_back(endMethodExClause);
+    newClauses.push_back(exClause);
+    newClauses.push_back(finallyClause);
+
+    return S_OK;
+}
+
+HRESULT DebuggerMethodRewriter::ApplyMethodSpanProbe(
+    ModuleID module_id, ModuleMetadata& module_metadata, FunctionInfo* caller, DebuggerTokens* debuggerTokens,
+    mdToken function_token, TypeSignature retFuncArg, bool isVoid, bool isStatic,
+    const std::vector<TypeSignature>& methodArguments, int numArgs,
+    const std::shared_ptr<SpanProbeOnMethodDefinition>& spanProbe,
+    ILRewriter& rewriter, const std::vector<TypeSignature>& methodLocals, int numLocals,
+    ILRewriterWrapper& rewriterWrapper, ULONG spanMethodStateIndex, ULONG exceptionIndex, ULONG callTargetReturnIndex,
+    ULONG returnValueIndex, mdToken callTargetReturnToken, const int instrumentedMethodIndex,
+    ILInstr*& beforeLineProbe, std::vector<EHClause>& newClauses) const
+{
+    const auto& spanProbeId = spanProbe->probeId;
+
+    LogDebugCallerInfo(caller, instrumentedMethodIndex);
+
+    rewriterWrapper.SetILPosition(beforeLineProbe);
+
+    const auto tryInstruction = beforeLineProbe->m_pPrev;
+
+    // ***
+    // BEGIN SPAN PART
+    // ***
+
+    // Define ResourceName as string
+    WSTRING resourceName =
+        spanProbe->target_method.type.name.substr(spanProbe->target_method.type.name.find_last_of(L'.') + 1) +
```

Comment:
out of curiosity: `L'.'` is valid on linux ?

Reply:
It's pretty much the same.
`#define WStr(value) L##value`.

Reply:
Done in 1bed3f68dfca764bd20a60fea21287b9274e0258.